### PR TITLE
Don't let inapplicable members participate in ORPA filtering

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/75871.

### CODE REVIEW

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1)

File: src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorAnalysisResult.cs [hunk 1]
```diff
@@ -36,6 +36,7 @@ public bool HasValue
             get { return this.Kind != OperatorAnalysisResultKind.Undefined; }
         }
 
+        bool IMemberResolutionResultWithPriority<MethodSymbol>.IsApplicable => IsValid;
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorAnalysisResult.cs:39]: nit: consider renaming existing `IsValid` property to `IsApplicable` instead. There's not much benefit of having two names for the same thing.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorAnalysisResult.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorAnalysisResult.cs:39]: I'd rather keep this scoped. If you want to come back with a refactoring of this I would be happy to review, but I don't want to conflate it with this change.

**Pull request merged**